lib/semian/simple_enum.rb

```diff
@@ -0,0 +1,29 @@
+module Semian
+  module Simple
+    class Enum #:nodoc:
+      def initialize(symbol_list:)
+        @allowed_symbols = symbol_list
+        reset
+      end
+
+      attr_reader :value
+
+      def value=(sym)
+        raise ArgumentError unless @allowed_symbols.include?(sym)
```

Comment:
Basically it's @byroot's suggestion. If I do:

```ruby
enum = Semian::Simple::Enum.new(symbol_list: [:one, :two, :three])
enum.value = :one
enum.value = :three
```

.. it works. I don't think that's what you want in this case.